### ISSUE TITLE
panic in ipfs block put

### ISSUE THREAD

**Issue by whyrusleeping** [jbenet/go-ipfs#767]
while working on examples I attempted to pipe data into the `ipfs block put` command and was greeted with this pleasant array of stacks:
```
(1) whyrusleeping@alarm ~> echo "HELP IM STUCK IN A BLOCK FACTORY" | ipfs block put
panic: interface conversion: interface is map[string]interface {}, not *commands.BlockStat

goroutine 1 [running]:
github.com/jbenet/go-ipfs/core/commands.funcÂ·008(0xb6ddb888, 0x10c2a1c0, 0x0, 0x0, 0x0, 0x0)
	/home/whyrusleeping/gopkg/src/github.com/jbenet/go-ipfs/core/commands/block.go:158 +0x84
github.com/jbenet/go-ipfs/commands.(*response).Marshal(0x10c2a1c0, 0x0, 0x0, 0x0, 0x0)
	/home/whyrusleeping/gopkg/src/github.com/jbenet/go-ipfs/commands/response.go:184 +0x64c
github.com/jbenet/go-ipfs/commands.(*response).Reader(0x10c2a1c0, 0x0, 0x0, 0x0, 0x0)
	/home/whyrusleeping/gopkg/src/github.com/jbenet/go-ipfs/commands/response.go:204 +0x104
main.(*cmdInvocation).Run(0x10c345c0, 0xb6dd9710, 0x10c345a0, 0x0, 0x0, 0x0, 0x0)
	/home/whyrusleeping/gopkg/src/github.com/jbenet/go-ipfs/cmd/ipfs/main.go:173 +0x2a0
main.main()
	/home/whyrusleeping/gopkg/src/github.com/jbenet/go-ipfs/cmd/ipfs/main.go:135 +0x578

goroutine 6 [syscall]:
os/signal.loop()
	/home/whyrusleeping/go/src/os/signal/signal_unix.go:21 +0x1c
created by os/signal.initÂ·1
	/home/whyrusleeping/go/src/os/signal/signal_unix.go:27 +0x38

goroutine 10 [chan receive]:
main.funcÂ·014()
	/home/whyrusleeping/gopkg/src/github.com/jbenet/go-ipfs/cmd/ipfs/main.go:479 +0x50
created by main.(*cmdInvocation).setupInterruptHandler
	/home/whyrusleeping/gopkg/src/github.com/jbenet/go-ipfs/cmd/ipfs/main.go:503 +0x130

goroutine 18 [IO wait]:
net.runtime_pollWait(0xb6ddb5e0, 0x72, 0x10c74301)
	/home/whyrusleeping/go/src/runtime/netpoll.go:157 +0x68
net.(*pollDesc).Wait(0x10c2a138, 0x72, 0x0, 0x0)
	/home/whyrusleeping/go/src/net/fd_poll_runtime.go:84 +0x3c
net.(*pollDesc).WaitRead(0x10c2a138, 0x0, 0x0)
	/home/whyrusleeping/go/src/net/fd_poll_runtime.go:89 +0x38
net.(*netFD).Read(0x10c2a100, 0x10ccf000, 0x1000, 0x1000, 0x0, 0xb6dd8ab0, 0x10c74390)
	/home/whyrusleeping/go/src/net/fd_unix.go:242 +0x368
net.(*conn).Read(0x10c1c7d0, 0x10ccf000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/home/whyrusleeping/go/src/net/net.go:121 +0xc8
net/http.noteEOFReader.Read(0xb6ddb658, 0x10c1c7d0, 0x10c9a02c, 0x10ccf000, 0x1000, 0x1000, 0x10c2a080, 0x0, 0x0)
	/home/whyrusleeping/go/src/net/http/transport.go:1276 +0x70
net/http.(*noteEOFReader).Read(0x10c74160, 0x10ccf000, 0x1000, 0x1000, 0x10c1e280, 0x0, 0x0)
	<autogenerated>:125 +0xec
bufio.(*Reader).fill(0x10ca60f0)
	/home/whyrusleeping/go/src/bufio/bufio.go:97 +0x1d8
bufio.(*Reader).Peek(0x10ca60f0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/whyrusleeping/go/src/bufio/bufio.go:132 +0xd4
net/http.(*persistConn).readLoop(0x10c9a000)
	/home/whyrusleeping/go/src/net/http/transport.go:842 +0x90
created by net/http.(*Transport).dialConn
	/home/whyrusleeping/go/src/net/http/transport.go:645 +0xb60

goroutine 19 [select]:
net/http.(*persistConn).writeLoop(0x10c9a000)
	/home/whyrusleeping/go/src/net/http/transport.go:945 +0x320
created by net/http.(*Transport).dialConn
	/home/whyrusleeping/go/src/net/http/transport.go:645 +0xb7c
```

**Comment by jbenet** [comment]
I think this is a problem in the commands lib not erroring out when it's not json?

**Comment by jbenet** [comment]
cc @mappum

**Comment by whyrusleeping** [comment]
block put shouldnt care about json, only the object/dag commands should care about json

**Comment by whyrusleeping** [comment]
@mappum fixed this as of #774
